ocs_ci/helpers/dr_helpers_ui.py

```diff
@@ -0,0 +1,274 @@
+
+    Args:
+        setup_acm_ui: login function to ACM on conftest file
+        down_cluster_name (str): If Failover is performed when a cluster goes down, wait is set to True & the updated
+                            status of cluster unavailability is checked on the ACM console.
+                            It takes the cluster name which is down.
+        cluster_names (list): This is a list of cluster names involved in a DR setup. You can either pass the cluster
+                            names as args in the form of list, but if not passed, it fetches the primary & secondary
+                            cluster names passed at run time for context setting
+                            (max. 3 for now including ACM Hub cluster).
+                            ACM Hub cluster name is hard coded as "local-cluster" as the name is constant & isn't
+                            expected to change.
+        timeout (int): Timeout to wait for certain elements to be found on the ACM UI
+        expected_text (str): Any particular string/status of the cluster to be checked on the ACM console.
+        wait (bool): When True, cluster status check will be done for the expected text/status in the given timeout.
+
+    """
+
+    ocp_version = version.get_semantic_ocp_version_from_config()
+    acm_obj = AcmPageNavigator(setup_acm_ui)
+    acm_loc = locators[ocp_version]["acm_page"]
+
+    acm_obj.navigate_clusters_page()
+
+    if down_cluster_name:
+        wait = True
+        check_cluster_unavailability = acm_obj.wait_until_expected_text_is_found(
+            format_locator(acm_loc["cluster_status_check"], down_cluster_name),
+            expected_text="Unavailable",
+            timeout=timeout,
+        )
+        if check_cluster_unavailability:
+            log.info(f"Down cluster {down_cluster_name} is {expected_text}")
+            return
+        else:
+            cluster_status = acm_obj.get_element_text(acm_loc["cluster_status_check"])
+            assert (
+                cluster_status == "Ready"
+            ), f"Down cluster {down_cluster_name} is still in {cluster_status} state after {timeout} seconds"
+            if not check_cluster_unavailability and cluster_status != "Ready":
+                other_expected_status = ["NotReady", "Error", "Unknown"]
+                for status in other_expected_status:
+                    check_cluster_unavailability = (
+                        acm_obj.wait_until_expected_text_is_found(
+                            format_locator(
+                                acm_loc["cluster_status_check"], down_cluster_name
+                            ),
+                            expected_text=status,
+                            timeout=30,
+                        )
+                    )
+                    if check_cluster_unavailability:
+                        f"Cluster {down_cluster_name} is in {status} state"
+                        return
```

Comment:
shouldn't we return the status or return a bool representing whether the cluster is in the desired state or not?

Reply:
Done